mayastor/src/bdev/nexus/nexus_label.rs

```diff
@@ -1269,121 +1304,82 @@ impl NexusChild {
 
-            let bdev = handle.get_bdev();
-            let label =
-                child.validate_label(&reference, bdev.block_len()).await?;
-            let data_blocks =
-                label.data_block_count().context(InvalidLabel {})?;
+            // Adjust size as necessary
+            size = min(size, label.partition_size("MayaData")?);
+        }
 
-            // Adjust size of data partition if necessary
-            if data_blocks < min_blocks {
-                min_blocks = data_blocks;
+        // Set the (common) "Data" offset
+        match unique(&offsets) {
+            Some(value) => {
+                self.data_ent_offset = value / block_size;
+            }
+            None => {
+                return Err(LabelError::DataOffsetMismatch {
+                    name: self.name.clone(),
+                });
             }
         }
 
-        // Update the nexus size
-        self.data_ent_offset = data_offset;
-        self.bdev.set_block_count(min_blocks);
+        // Set the nexus size
+        self.bdev.set_block_count(size / block_size);
 
         Ok(())
     }
 
-    // Get configuration from first valid label with specified disk guid
-    async fn find_label_config(
-        &self,
-        guid: GptGuid,
-    ) -> Result<Option<LabelConfig>, LabelError> {
-        for child in self.children.iter() {
-            match child.probe_label().await {
-                Ok(label) => {
-                    if label.primary.guid != guid {
-                        continue;
-                    }
-                    if let Some(config) = label.get_label_config() {
-                        return Ok(Some(config));
-                    }
-                }
-                Err(LabelError::InvalidLabel {
-                    ..
-                }) => {
-                    // Label is most likely not present or possibly invalid.
-                    continue;
-                }
-                Err(error) => {
-                    // Any other errors are fatal.
-                    return Err(error);
-                }
-            }
-        }
-        Ok(None)
-    }
-
     /// Create or Update label on each child device as and when necessary
     pub(crate) async fn update_child_labels(
         &mut self,
     ) -> Result<(), LabelError> {
-        let guid = GptGuid::from(Uuid::from_bytes(self.bdev.uuid().as_bytes()));
-        let config = self
-            .find_label_config(guid)
-            .await?
-            .unwrap_or_else(|| LabelConfig::new(guid));
-
-        let block_size = self.bdev.block_len();
-        let nexus_blocks = self.size / u64::from(block_size);
-
-        // Generate "reference" partition table entries
-        let header = GptHeader::reference(block_size, nexus_blocks, guid);
-        let reference = Nexus::create_maya_partitions(
-            &config,
-            &header,
-            block_size,
-            nexus_blocks,
-        )?;
+        let now = SystemTime::now();
```

Comment:
I am a bit worried how using SystemTime might interact over rebuilds with remotes. Do we need to worry about cases where one machine has a very different time than the other?